test/d2l-image.test.js

```diff
@@ -0,0 +1,84 @@
+import '../d2l-image.js';
+import { expect, fixture, html, oneEvent } from '@open-wc/testing';
+import { stub } from 'sinon';
+
+describe('d2l-image', () => {
+	let fetchStub;
+
+	beforeEach(() => {
+		fetchStub = stub(window, 'fetch');
+	});
+
+	afterEach(() => {
+		fetchStub.restore();
+	});
+
+	it('loads element', async() => {
+		const el = await fixture(html`<d2l-image></d2l-image>`);
+		expect(el).to.exist;
+	});
+
+	it('does not render an <img> without image url', async() => {
+		const el = await fixture(html`<d2l-image></d2l-image>`);
+		expect(el.shadowRoot.querySelector('img')).to.not.exist;
+	});
+
+	describe('Hypermedia calls', () => {
+
+		beforeEach(() => {
+			fetchStub.resolves({ ok: true });
+		});
+
+		describe('Sending requests', () => {
+
+			it('should not send a request without an image url', async() => {
+				const el = await fixture(html`<d2l-image></d2l-image>`);
+
+				expect(el.imageUrl).to.be.undefined;
+				expect(fetchStub).to.not.have.been.called;
+			});
+
+			it('should send a request with an image url', async() => {
+				const el = await fixture(html`<d2l-image image-url="image.jpg"></d2l-image>`);
+
+				expect(el.imageUrl).to.equal('image.jpg');
+				expect(fetchStub).to.have.been.called;
+			});
+		});
+
+		describe('Receiving responses', () => {
+
+			it('should update image src attribute and fire image-loaded event if request was successful', async() => {
+				const response = new Response(new Blob(['image'], { type: 'image/jpeg' }));
+				fetchStub.resolves(response);
+
+				const el = await fixture(html`<d2l-image></d2l-image>`);
+
+				el.imageUrl = 'image4.jpg';
```

Comment:
If you want to avoid a potential race condition where the event fires before your `await oneEvent` call, you can wrap this in a timeout:

```suggestion
				setTimeout(() => el.imageUrl = 'image4.jpg');
```